### DISCUSSION TITLE
How to lock the TALTabControl move Horizontally and TALVertScrollBox move vertically

### DISCUSSION THREAD

**Discussion by pejuangdelphi** [MagicFoundation/Alcinoe#188] [Show and tell]
@Zeus64

I have installed the app KisKis, it's so cool and so smooth...

![image](https://user-images.githubusercontent.com/45804972/118739532-e1901c80-b873-11eb-9a1a-b7c3db55e47b.png)

based on the picture above, when moving up or down, the app locks the screen to move left or right. on another side, when moving to left or right, the apps then locking to move up or down.

I have attached my demo project, please let me know how to achieve like app KisKis

Thank in advanced

[Demo TALTabControl.zip](https://github.com/Zeus64/alcinoe/files/6505202/Demo.TALTabControl.zip)

**Comment by pejuangdelphi** [comment]
@Zeus64
Please let me knows when the explanation above is not clear...

**Comment by Zeus64** [comment]
Hello, it's quite complicated to explain, basically their is 2 TALAniCalculation, one for the vertical, and one for the horizontal. when the horizontal start you lock the vertical, and when the vertical start you lock the horizontal. their is also a dead zone (around 5 pixels if i remember well) and you lock only after one of the 2 anicalculation move more pixel that the one in the deadzone

**Reply by pejuangdelphi** [reply]
> 
> 
> Hello, it's quite complicated to explain, basically their is 2 TALAniCalculation, one for the vertical, and one for the horizontal. when the horizontal start you lock the vertical, and when the vertical start you lock the horizontal. their is also a dead zone (around 5 pixels if i remember well) and you lock only after one of the 2 anicalculation move more pixel that the one in the deadzone

Thank you so much for the replies.

Understandable. I think so. But, could you help me to make it simple demo for it based on my attachment before?

I will try to learn it...

Please help me in order to have an application like KisKis... can move horizontal then locking the vertical

**Reply by pejuangdelphi** [reply]
@Zeus64
I try to look for the class TALAniCalculation and can't find it...

Please help me to have the demo for it... or if have to pay for it. please let me know since i have to finish my project.
